diff --git a/src/cre2.cpp b/src/cre2.cpp
index 5a63b93..78d92c3 100644
--- a/src/cre2.cpp
+++ b/src/cre2.cpp
@@ -216,7 +216,7 @@ cre2_program_size (const cre2_regexp_t *re)
 struct cre2_named_groups_iter_t
 {
   const RE2 * re;
-  std::map<std::string, int>::const_iterator it;
+  std::map<int, std::string>::const_iterator it;
 };
 
 cre2_named_groups_iter_t *
@@ -224,19 +224,19 @@ cre2_named_groups_iter_new(const cre2_regexp_t *re)
 {
   cre2_named_groups_iter_t * iter = new (std::nothrow)cre2_named_groups_iter_t;
   iter->re = TO_CONST_RE2(re);
-  iter->it = iter->re->NamedCapturingGroups().begin();
+  iter->it = iter->re->CapturingGroupNames().begin();
   return iter;
 }
 bool
 cre2_named_groups_iter_next(cre2_named_groups_iter_t* iter, const char ** name, int *index)
 {
-  if (iter->it == iter->re->NamedCapturingGroups().end()) {
+  if (iter->it == iter->re->CapturingGroupNames().end()) {
     *name = NULL;
     *index = -1;
     return false;
   } else {
-    *index = iter->it->second;
-    *name = iter->it->first.c_str();
+    *index = iter->it->first;
+    *name = iter->it->second.c_str();
     ++iter->it;
     return true;
   }
@@ -462,7 +462,6 @@ DEFINE_MATCH_REX_FUN2(cre2_find_and_consume_re,FindAndConsumeN)
 int
 cre2_replace (const char * pattern, cre2_string_t * text_and_target, cre2_string_t * rewrite)
 {
-  try {
     std::string		S(text_and_target->data, text_and_target->length);
     re2::StringPiece	R(rewrite->data, rewrite->length);
     char *		buffer; /* this exists to make GCC shut up about const */
@@ -477,12 +476,6 @@ cre2_replace (const char * pattern, cre2_string_t * text_and_target, cre2_string
     } else
       return -1;
     return int(retval);
-  } catch(const std::exception &e) {
-    // e.what();
-    return -1;
-  } catch(...) {
-    return -1;
-  }
 }
 int
 cre2_replace_re (cre2_regexp_t * rex, cre2_string_t * text_and_target, cre2_string_t * rewrite)
