diff --git a/src/cre2.cpp b/src/cre2.cpp
index 5a63b93..54152ff 100644
--- a/src/cre2.cpp
+++ b/src/cre2.cpp
@@ -288,6 +288,27 @@ cre2_match (const cre2_regexp_t *re , const char *text,
   }
   return (retval)? 1 : 0;
 }
+
+int
+cre2_match8 (const cre2_regexp_t *re , const char *text,
+            int textlen, int startpos, int endpos, cre2_anchor_t anchor,
+            cre2_string_t *match, int nmatch)
+{
+  re2::StringPiece	text_re2(text, textlen);
+  re2::StringPiece match_re2[8];
+  RE2::Anchor		anchor_re2 = to_cre2_anchor(anchor);
+  bool			retval; // 0 for no match
+  // 1 for successful matching
+  retval = TO_CONST_RE2(re)->Match(text_re2, startpos, endpos, anchor_re2, match_re2, 8);
+  if (retval) {
+    for (int i=0; i<8; i++) {
+      match[i].data   = match_re2[i].data();
+      match[i].length = match_re2[i].length();
+    }
+  }
+  return (retval)? 1 : 0;
+}
+
 int
 cre2_easy_match (const char * pattern, int pattern_len,
 		 const char *text, int text_len,
@@ -462,7 +483,6 @@ DEFINE_MATCH_REX_FUN2(cre2_find_and_consume_re,FindAndConsumeN)
 int
 cre2_replace (const char * pattern, cre2_string_t * text_and_target, cre2_string_t * rewrite)
 {
-  try {
     std::string		S(text_and_target->data, text_and_target->length);
     re2::StringPiece	R(rewrite->data, rewrite->length);
     char *		buffer; /* this exists to make GCC shut up about const */
@@ -477,12 +497,6 @@ cre2_replace (const char * pattern, cre2_string_t * text_and_target, cre2_string
     } else
       return -1;
     return int(retval);
-  } catch(const std::exception &e) {
-    // e.what();
-    return -1;
-  } catch(...) {
-    return -1;
-  }
 }
 int
 cre2_replace_re (cre2_regexp_t * rex, cre2_string_t * text_and_target, cre2_string_t * rewrite)
diff --git a/src/cre2.h b/src/cre2.h
index 92eaf65..a34051f 100644
--- a/src/cre2.h
+++ b/src/cre2.h
@@ -160,6 +160,11 @@ cre2_decl int cre2_match	(const cre2_regexp_t * re,
 				 int startpos, int endpos, cre2_anchor_t anchor,
 				 cre2_string_t * match, int nmatch);
 
+cre2_decl int cre2_match8	(const cre2_regexp_t * re,
+                             const char * text, int textlen,
+                             int startpos, int endpos, cre2_anchor_t anchor,
+                             cre2_string_t * match, int nmatch);
+
 cre2_decl int cre2_easy_match	(const char * pattern, int pattern_len,
 				 const char * text, int text_len,
 				 cre2_string_t * match, int nmatch);
